modules/installation-adding-registry-pull-secret.adoc

```diff
@@ -136,59 +131,74 @@ $ cp <path>/<pull_secret_file_in_json> <directory_name>/<auth_file>
 ----
 +
 Where `<directory_name>` is either `~/.docker` or `$XDG_RUNTIME_DIR/containers`, and `<auth_file>` is either `config.json` or `auth.json`.
-endif::[]
+endif::update-oc-mirror[]
 // Additional step for allowing this procedure for oc-mirror-v2
+// Should this step below also have the "if you don't have this directory, create it using this command" substeps?
 ifdef::oc-mirror-v2[]
-. Save the file as `$XDG_RUNTIME_DIR/containers/auth.json`.
-endif::[]
-endif::[]
- 
-. Generate the base64-encoded user name and password or token for your mirror registry:
+. If the `$XDG_RUNTIME_DIR/containers` directory does not exist, create one by entering the following command:
++
+[source,terminal]
+----
+$ mkdir -p $XDG_RUNTIME_DIR/containers
+----
+
+. Save the pull secret file as `$XDG_RUNTIME_DIR/containers/auth.json`.
+endif::oc-mirror-v2[]
+endif::openshift-origin[]
+
+. Generate the base64-encoded user name and password or token for your mirror registry by running the following command:
 +
 [source,terminal]
 ----
 $ echo -n '<user_name>:<password>' | base64 -w0 <1>
-BGVtbYk3ZHAtqXs=
 ----
 <1> For `<user_name>` and `<password>`, specify the user name and password that you configured for your registry.
++
+.Example output
+[source,terminal]
+----
+BGVtbYk3ZHAtqXs=
+----
 
-ifndef::openshift-origin[]
-. Edit the JSON
-endif::[]
 ifdef::openshift-origin[]
-. Create a `.json`
-endif::[]
-file and add a section that describes your registry to it:
+. Create a `.json` file and add a section that describes your registry to it:
 +
 [source,json]
 ----
-ifndef::openshift-origin[]
+{
   "auths": {
     "<mirror_registry>": { <1>
       "auth": "<credentials>", <2>
       "email": "you@example.com"
     }
-  },
-endif::[]
-ifdef::openshift-origin[]
-{
+  }
+}
+----
+<1> Specify the registry domain name, and optionally the port, that your mirror registry uses to serve content. For example,
+`registry.example.com` or `registry.example.com:8443`
+<2> Specify the base64-encoded user name and password for
+the mirror registry.
+
+endif::openshift-origin[]
+
+ifndef::openshift-origin[]
+. Edit the JSON file and add a section that describes your registry to it:
```

Comment:
Does line.no:164 and line no:185 duplicates ? because i see both of them describes procedure about `registry describe`

Reply:
They are similar, but lines 164-180 are conditionalized to only show up on the OKD version of this procedure, and lines 185-228 show up in every other distribution (that uses this file). That's what the ifdefs and ifndefs such as the ones on line 184 are doing